R/eatModel.r

```diff
@@ -1,94 +1,59 @@
+    x <- eatTools::crop(paste(x, sep = "", collapse = ""))
+    x <- unlist(strsplit(x, " +"))
+    return(x)
+  })
+  return(isL)
+}
 
 ### not called ----------------------------------------------------------------------------
 
-simEquiTable <- function ( anchor, mRef, sdRef, addConst = 500, multConst = 100, cutScores) {
-                anchor<- eatTools::makeDataFrame(anchor)
-                if ( ncol(anchor) != 2) {
-                     warning(paste0("'anchor' has ",ncol(anchor)," columns. First column is used as item ID, second column is used as item parameter."))
-                }
-                if(!inherits(anchor[,2], c("integer", "numeric"))) {stop("Item parameter column must be numeric.")}
-                if(length(unique(anchor[,1])) != nrow(anchor)) {stop("Item ID column has duplicated entries.")}
-                dtmp  <- data.frame(rbind(1*(lower.tri(matrix(1, nrow = nrow(anchor), ncol = nrow(anchor)))),1))
-                dtmp  <- data.frame(dtmp, score = rowSums(dtmp) , irtoys::wle(dtmp, cbind(1, anchor[,2], 0)), stringsAsFactors = FALSE)
-                dtmp[,"bista"] <- (dtmp[,"est"] - mRef) / sdRef * multConst + addConst
-                dtmp[,"ks"]    <- eatTools::num.to.cat ( x = dtmp[,"bista"], cut.points = cutScores[["values"]], cat.values = cutScores[["labels"]])
-                dig   <- 0
-                while ( length(which(round(dtmp[,"bista"], digits = dig) %in% cutScores[["values"]])) > 0) {dig <- dig + 1}
-                shrt  <- do.call("rbind", by ( data = dtmp, INDICES = dtmp[,"ks"], FUN = function ( sks ) { data.frame ( score = paste(c(min(sks[,"score"]), max(sks[,"score"])), collapse=" bis "), estimate = paste(round(c(min(sks[,"est"]), max(sks[,"est"])),digits=2), collapse=" bis "), bista = paste(round(c(min(sks[,"bista"]), max(sks[,"bista"])),digits=dig), collapse=" bis "), ks=unique(sks[,"ks"]), stringsAsFactors=FALSE)}))
-                return(list ( complete = dtmp[,c("score", "est", "bista", "ks")], short = shrt))}
+simEquiTable <- function(anchor, mRef, sdRef, addConst = 500, mulConst = 100, cutScores){
+  # checks/prep
+  anchor <- eatTools::makeDataFrame(anchor)
+  if(ncol(anchor) != 2){
+    warning(paste0("'anchor' has ", ncol(anchor)," columns. First column is used as item ID, second column is used as item parameter."))
+  }
+  if(!inherits(anchor[,2], c("integer", "numeric"))){
+    stop("Item parameter column must be numeric.")
+  }
+  if(length(unique(anchor[,1])) != nrow(anchor)){
+    stop("Item ID column has duplicated entries.")
+  }
+  checkmate::assert_numeric(mRef, len = 1)
+  lapply(c(sdRef, addConst, mulConst), checkmate::assert_numeric, len = 1, lower = 0)
+  checkmate::assert_list(cutScores, min.len = 1, max.len = 2)
+  checkmate::assert_numeric(cutScores$values, sorted = TRUE)
+  checkmate::assert_character(cutScores$labels, null.ok = TRUE, len = length(cutScores$values),
+                              sorted = TRUE, unique = TRUE)

```

Comment:
sorted works for character input as well. It gives an error message when the labels are c("I", "II", "IV", "III") or c("A1", "A2", "C1", "B1", ...) - so if the letters are in wrong order OR the numbers - but the message only says "must be ordered" and not where the error is or how to order that. 
I could just write a custom error message for that, though, with an if statement; or drop that altogether.

Reply:
Length of `cutScores$labels` must be `length(cutScores$values) + 1`, i.e. for only one cut score (say, 100), you have two labels, for example, `"above"` and `"below"`. I would suggest to use `sorted = FALSE`, because at least `labels = c("I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX", "X")` will give an error although it's a valid statement.